application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,413 @@
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null or a {@link DateTimeParseException} is caught, the
+     * oldest-possible {@link ZonedDateTime} will get returned instead.
+     *
+     * @param item The {@link Item} from which to extract the date.
+     * @param dateTimeFormat The format of the date time string.
+     * @return The computed {@link ZonedDateTime}
+     * @throws DateTimeParseException if the date cannot be parsed
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat)
+            throws DateTimeParseException {
+        String pubDate = item.getPubDate().orElse(null);
+
+        if (pubDate == null || dateTimeFormat == null) {
+            return ZONED_TIME_MIN;
+        }
+
+        return getZonedDateTime(pubDate, dateTimeFormat);
+    }
+
+    /**
+     * Checks if the dates between an RSS item and the provided config match.
+     *
+     * @param rssItem the RSS feed item
+     * @param feedConfig the RSS feed configuration containing the date formatter pattern
+     * @return true if the date format is valid, false otherwise
+     */
+    private static boolean isValidDateFormat(Item rssItem, RSSFeed feedConfig) {
+        String firstRssFeedPubDate = rssItem.getPubDate().orElse(null);
+
+        if (firstRssFeedPubDate == null) {
+            return false;
+        }
+
+        try {
+            // If this throws a DateTimeParseException then it's certain
+            // that the format pattern defined in the config and the
+            // feed's actual format differ.
+            getZonedDateTime(firstRssFeedPubDate, feedConfig.dateFormatterPattern());
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to find a text channel from a given RSS feed configuration.
+     *
+     * @param jda the JDA instance
+     * @param feed the RSS feed configuration to search for a text channel
+     * @return an {@link Optional} containing the found text channel if it exists, otherwise empty
+     */
+    private Optional<TextChannel> getTextChannelFromFeed(JDA jda, RSSFeed feed) {
+        // Attempt to find the target channel, use the fallback otherwise
+        if (targetChannelPatterns.containsKey(feed)) {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> targetChannelPatterns.get(feed).test(channel.getName()))
+                .findFirst();
```

Comment:
Shouldn't it post in all matching channels?

Reply:
There is only one channel that it can be posted in. It is not the intended behavior to be able to post in multiple channels.

Reply:
> There is only one channel that it can be posted in.

How that?

> It is not the intended behavior to be able to post in multiple channels.

Why not? The pattern can match multiple channels.

Reply:
Done!